subsys/net/lib/sockets/sockets.c

```diff
@@ -84,14 +85,34 @@ int zsock_socket(int family, int type, int proto)
 	/* recv_q and accept_q are in union */
 	k_fifo_init(&ctx->recv_q);
 
+#ifdef CONFIG_USERSPACE
+	/* Set net context object as initialized and grant access to the
+	 * calling thread (and only the calling thread)
```

Comment:
> (and only the calling thread)

I love that - no I/O resource sharing between threads, kids! ;-)

Reply:
That's not how it works.

Additional threads may be granted access to the context object with k_object_access_grant(). This is just the default policy for newly acquired objects.